src/libsyntax/test.rs

```diff
@@ -690,19 +733,65 @@ fn mk_test_desc_and_fn_rec(cx: &TestCtxt, test: &Test) -> P<ast::Expr> {
              field("should_panic", fail_expr),
              field("allow_fail", allow_fail_expr)]);
 
-
-    let mut visible_path = match cx.toplevel_reexport {
-        Some(id) => vec![id],
+    let mut visible_path = vec![];
+    if cx.features.extern_absolute_paths {
+        visible_path.push(keywords::Crate.ident());
+    }
+    match cx.toplevel_reexport {
+        Some(id) => visible_path.push(id),
         None => {
             let diag = cx.span_diagnostic;
             diag.bug("expected to find top-level re-export name, but found None");
         }
     };
     visible_path.extend(path);
 
-    let fn_expr = ecx.expr_path(ecx.path_global(span, visible_path));
+    // If termination feature is enabled, create a wrapper that invokes the fn
+    // like this:
+    //
+    //     fn wrapper() {
+    //         assert_eq!(0, real_function().report());
```

Comment:
sure you mean `Termination::assert_unit_test_successful(real_function())`?

Reply:
yeah that comment is dated I guess